src/main/java/patching/Diff.java

```diff
@@ -179,12 +179,8 @@ public Diff getUndo() {
 
                 // CASE 1: IndexA < IndexB
                 if (other.startIndex < current.startIndex) {
-                    // CASE 1a: Ins - Ins
-                    if (other.insertion && current.insertion) {
-                        transformType2(newIntermediateDiffs, current, other);
-                    }
-                    // CASE 1b: Ins - Rmv
-                    else if (other.insertion && !current.insertion) {
+                    // CASE 1a, 1b: Ins - Ins, Ins - Rmv
+                    if ((other.insertion && current.insertion) || (other.insertion && !current.insertion)) {
```

Comment:
I get that this is trying to be more verbose, but now this just looks like a silly statement to have in a strongly typed language...